go/libraries/doltcore/sqle/stats/update.go

```diff
@@ -200,6 +174,58 @@ func refreshStats(ctx *sql.Context, indexes []sql.Index, idxMetas []indexMeta) (
 	return ret, nil
 }
 
+func mergeStatUpdates(newStats *DoltStats, idxMeta indexMeta) *DoltStats {
+	if len(newStats.Histogram) == len(idxMeta.allAddrs) {
+		newStats.updateActive()
+		return newStats
+	}
+	oldHist := idxMeta.preexisting
+	var mergeHist DoltHistogram
+	newHist := newStats.Histogram
+	var i, j int
+	for _, chunkAddr := range idxMeta.allAddrs {
+		if i < len(oldHist) && oldHist[i].Chunk == chunkAddr {
+			mergeHist = append(mergeHist, oldHist[i])
+			i++
+		} else if j < len(newHist) && newHist[j].Chunk == chunkAddr {
```

Comment:
Possible bug: if an identical chunk appears in both the old and new histograms, then j will never be incremented and all the new chunks that come after it will be skipped.

Reply:
yes this was a bug